src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -118,6 +118,34 @@ export class NotebookLinkHandler {
 		}
 	}
 
+	/**
+	* Function to get encoded url,
+	* checks the url if it's already encoded before encoding it
+	* @returns the encoded url
+	*/
+	public getEncodedLinkUrl(): string | undefined {
+		// since we only handle strings that come from call out dialogs
+		if (typeof this._link === 'string') {
+			// Need to encode URI here in order for user to click the proper encoded link in WYSIWYG
+			// skip encoding it if it's already encoded
+			if (!this.containsEncodedComponents(this._link)) {
+				return encodeURI(this._link);
+			}
+			return this._link;
+		}
+		return undefined;
+	}
+
+	/**
+	* Function to check if a string url has encoded characters
```

Comment:
```
Checks if the specified URL is already URI-encoded by checking if there are any uncoded reserved URI component characters (such as ?, =, &, /, etc.) in the URL. See https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent for more details.
@returns true if the URL contains encoded URI component reserved characters
```

I'd also make this clearer and update the signature to : 

`containsEncodedUriComponentReservedCharacters(url: string): boolean`

Reply:
And as mentioned I'd suggest making a new file at `src/sql/base/common/network.ts` and putting this in there so others can find it more easily.